### ISSUE TITLE
Content Element with uid "xxxx" and type "xxxx" has no rendering definition!

### ISSUE THREAD

**Issue by Fanor51** [TYPO3-Headless/headless#627]
Hello,

i am upgrading a TYPO3 v11 with the headless extensions to v12 and the latest headless version.

Before: 
PHP: 8.0
TYPO3 : v11.5.26
headless  :3.3.1

After: 
PHP: 8.2
TYPO3 : v12.4.3
headless : v4.1.0

I extended the typoscript from a Textmedia with a new field. Copied the config from the headless extensions and import the new one in my extensions like:
![image](https://github.com/TYPO3-Headless/headless/assets/10448557/64d27162-96cd-4b78-8887-8a12226c8432)

The file is imported after your extension so there should be no loading order error. 
![image](https://github.com/TYPO3-Headless/headless/assets/10448557/e6f549ac-5865-4360-b73a-a2ce28e06f3d)

The API shows me this error:
![image](https://github.com/TYPO3-Headless/headless/assets/10448557/fefe6346-5f3e-4620-ad03-72498660494e)

On the "old" TYPO3 system this works all with no problem. On both (Headless, TYPO3) changelogs i cant find anything which this could be the issue. When i delete the complete content from my overwritten ts file the module is shown normaly but when I copy the file one to one the error occurs too. 

I am a bit stucked and maybe you can re-produce or find a bug some where. My next step is to dive in to the headless core and maybe find something.

Maybe you guys have an idea what this could be.

Best Regards and thanks!
Fabio

**Comment by twoldanski** [comment]
Hi @Fanor51 you copied old version of `textmedia` definition, unfortunately TYPO3 v12 and new typoscript parser has some regressions. Your issue is caused by line `tt_content.textmedia  < lib.baseImage` (that's why we temporarily removed id in 4.x version).

**Comment by tmotyl** [comment]
@twoldanski isn't the regression fixed already in current v12?
Maybe we could bump the minimal req for headless to make sure the fix is already there and simplyfi the ts again?

**Comment by twoldanski** [comment]
@tmotyl No, it's not fixed. I tested on TYPO3 `12.4.3`

**Comment by Fanor51** [comment]
Hello @twoldanski & @tmotyl,

first of all thanks for your support. @twoldanski what do you mean with "removed temporarily"? I still can see it in the extension or do i miss a point? When i delete the lib.baseImage line i get an other error: 
![image](https://github.com/TYPO3-Headless/headless/assets/10448557/2b7e8bb3-7977-4861-a1dc-df75cb9532b3)

and do i understand this right that atm there is no fix for this and if have just to wait or can i help somewhere to accelerate this fix?

Thank you both!

Fabio

**Comment by lukaszuznanski** [comment]
@Fanor51 you need to migrate your own definition of textmedia. 

Line that is causing problem is:
```
tt_content.textmedia  < lib.baseImage
```

Take a look at typoscript in master or 4.x branch
https://github.com/TYPO3-Headless/headless/blob/master/Configuration/TypoScript/ContentElement/Textmedia.typoscript


And that is commit that fixed this regression.

https://github.com/TYPO3-Headless/headless/commit/2f38692b348017e9a27724959678514a197d1798

This is bug in core (they changed typoscript parser and now it doesn't support this behaviour, it's in forge issue that @twoldanski posted above.
You should migrate your elements like in commit provided above.

**Comment by Fanor51** [comment]
@lukaszuznanski  Alright i got it and it works but only without my new field:

```
collapsibleBodytext = TEXT
collapsibleBodytext {
  field = collapsible_bodytext
  parseFunc =< lib.parseFunc_links
}
```

idk if this is another topic but when i call this element without `parseFunc =< lib.parseFunc_links` content is shown:
![image](https://github.com/TYPO3-Headless/headless/assets/10448557/4909e240-0963-4d6c-8ec9-eee108539c07)
with it i get the error above (https://github.com/TYPO3-Headless/headless/issues/627#issuecomment-1643482809). Are there any new changes in the lib function? 

If you want i can make another issue of this is a bug. 

Thank you guys for your help!

Fabio

**Comment by twoldanski** [comment]
@Fanor51 hmm I'm unable to reproduce it. what configuration & content do you have this error ?

**Comment by Fanor51** [comment]
@twoldanski i got this.

it occurs when the db field is NULL:

![image](https://github.com/TYPO3-Headless/headless/assets/10448557/d36a8f03-b298-46b7-82c2-212e84e0370b)

My table config looks like this '	collapsible_bodytext mediumtext' and i think i need to set an empty string as default value. At the moment if the field is not set, it sets the value 'NULL' in it.

But the core got this like this as well : 

![image](https://github.com/TYPO3-Headless/headless/assets/10448557/8753c115-d018-45c1-8c0a-eb0ac74d55a6)

Maybe a TCA setting is missing. But jeah, it looks like an error on my site. 

Thank you!

**Comment by twoldanski** [comment]
@Fanor51 Yeah, your confirmed my thoughts, you have have default value set properly, and main issue is still in core TYPO3 not in ext:headless

**Comment by lukaszuznanski** [comment]
Closing for now, @Fanor51 please reopen if you have any additional questions,

thanks!

**Comment by tmotyl** [comment]
good news, Thanks to Tomek and Lolli the patch was merged to the core https://review.typo3.org/c/Packages/TYPO3.CMS/+/80461 so it will be released in 12.4.6
